Framework/Parallel/src/StorageMode.cpp

```diff
@@ -25,5 +25,18 @@ std::string toString(const std::map<std::string, StorageMode> &map) {
   return ret;
 }
 
+/// Returns a StorageMode for a human-readable string representation.
+StorageMode fromString(const std::string &mode) {
+  if (mode == "Parallel::StorageMode::Cloned")
```

Comment:
Is there a reason you use, for example, `Parallel::StorageMode::Cloned` instead of just `Cloned`?